From: nabijaczleweli@nabijaczleweli.xyz
Date: Wed, 17 Aug 2022 22:33:38 +0200
Subject: Link against libncursesw, setlocale(LC_ALL, "")

On https://example.com/a” (i.e. a fancy quote), urlview with
narrow-character curses makes an absolute hash of it, by writing it as:
  https://example.com/a�~@~]

When linked against the wide-character ncurses, it displays it exactly
as expected, in the original form.
---
diff --git a/configure.in b/configure.in
index 45e4e89..7f5be5c 100644
--- a/configure.in
+++ b/configure.in
@@ -24,10 +24,10 @@ AC_ARG_WITH(slang, [  --with-slang[=DIR]      use S-Lang instead of curses],
 		LIBS="$LIBS -lslang -lm"
 	fi],
 
-	[AC_CHECK_LIB(ncurses, initscr, [LIBS="$LIBS -lncurses"], [LIBS="$LIBS -lcurses"])
+	[AC_CHECK_LIB(ncursesw, initscr, [LIBS="$LIBS -lncursesw"], [LIBS="$LIBS -lcurses"])
 	AC_CHECK_FUNCS(curs_set)
 	AC_CHECK_HEADERS(ncurses.h)
-	AC_CHECK_HEADER(ncurses/curses.h, [CPPFLAGS="$CPPFLAGS -I/usr/include/ncurses"])])
+	AC_CHECK_HEADER(ncursesw/curses.h, [CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"])])
 
 AC_REPLACE_FUNCS(snprintf)
 
diff --git a/urlview.c b/urlview.c
index 7416dfc..b0861bd 100644
--- a/urlview.c
+++ b/urlview.c
@@ -32,6 +32,7 @@
 #endif /* USE_SLANG */
 
 #include <pwd.h>
+#include <locale.h>
 #include <string.h>
 #include <unistd.h>
 #include <ctype.h>
@@ -146,6 +147,8 @@ void search_backward (char *search, int urlcount, char **url, int *redraw, int *
  
 int main (int argc, char **argv)
 {
+  setlocale(LC_ALL, "");
+
   struct passwd *pw;
   struct stat stat_buf;
 #ifndef USE_SLANG
